src/lib/io/sol-ipm.c

```diff
@@ -0,0 +1,323 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdlib.h>
+
+#ifndef SOL_LOG_DOMAIN
+#define SOL_LOG_DOMAIN &_sol_ipm_log_domain
+#endif
+
+#include "sol-log-internal.h"
+
+#include "ipm.h"
+#include "ipm/ipm_quark_se.h"
+
+#include "sol-log.h"
+#include "sol-macros.h"
+#include "sol-mainloop-zephyr.h"
+#include "sol-types.h"
+#include "sol-util.h"
+#include "sol-vector.h"
+
+SOL_LOG_INTERNAL_DECLARE_STATIC(_sol_ipm_log_domain, "ipm");
+
+#define SOL_IPM_ID_FREE_REMOTE 0
+
+/* Declare channels we are going to use.
+ * Channel 0 is Quark -> ARC
+ * Channel 1 is ARC -> Quark
+ */
+#ifdef CONFIG_X86
+QUARK_SE_IPM_DEFINE(message_ipm_receiver, 1, QUARK_SE_IPM_INBOUND);
+QUARK_SE_IPM_DEFINE(message_ipm_sender, 0, QUARK_SE_IPM_OUTBOUND);
+#elif CONFIG_ARC
+#warning "Building for ARC"
+QUARK_SE_IPM_DEFINE(message_ipm_receiver, 0, QUARK_SE_IPM_INBOUND);
+QUARK_SE_IPM_DEFINE(message_ipm_sender, 1, QUARK_SE_IPM_OUTBOUND);
+#endif
+
+/*XXX there MUST be a better way of doing so */
+#define AS_PTR(_ptr) *(uint64_t *)_ptr
```

Comment:
On 05-05-2016 18:41, Gustavo Sverzut Barbieri wrote:
> In src/lib/io/sol-ipm.c
> <https://github.com/solettaproject/soletta/pull/1981#discussion_r62260174>:
>
>> +
>> +/* Declare channels we are going to use.
>> + * Channel 0 is Quark -> ARC
>> + * Channel 1 is ARC -> Quark
>> + */
>> +#ifdef CONFIG_X86
>> +QUARK_SE_IPM_DEFINE(message_ipm_receiver, 1, QUARK_SE_IPM_INBOUND);
>> +QUARK_SE_IPM_DEFINE(message_ipm_sender, 0, QUARK_SE_IPM_OUTBOUND);
>> +#elif CONFIG_ARC
>> +#warning "Building for ARC"
>> +QUARK_SE_IPM_DEFINE(message_ipm_receiver, 0, QUARK_SE_IPM_INBOUND);
>> +QUARK_SE_IPM_DEFINE(message_ipm_sender, 1, QUARK_SE_IPM_OUTBOUND);
>> +#endif
>> +
>> +/*XXX there MUST be a better way of doing so */
>> +#define AS_PTR(_ptr) *(uint64_t *)_ptr
>
> |uintptr_t|?

I swear I was trying to find such a type... tks!

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1981/files/9efa022b5f0960e84101624c9866a4cd1a343cbb#r62260174>
>